src/psyclone/psyir/nodes/array_member.py

```diff
@@ -77,8 +77,12 @@ def create(member_name, indices, parent=None):
         :type parent: subclass of :py:class:`psyclone.psyir.nodes.Node`
 
         '''
-        return ArrayMember._create_array_member(
-            member_name, indices=indices, parent=parent)
+        obj = ArrayMember(member_name, parent=parent)
+        # Add any array-index expressions as children
+        for child in indices:
+            obj.addchild(child)
+            child.parent = obj
+        return obj
 
     @staticmethod
     def _validate_child(position, child):
```

Comment:
Can we move this (and ArrayReferece._validate_children) inside ArrayMixin now that only these 2 classes use it directly?

Reply:
Oh yes, that's nicer. I also realised I could move reference_accesses too. I had a bit of trouble with the MRO so had to alter the order with which the classes inherit from e.g. `Member` and `ArrayMixin`.